code/modules/reagents/chemistry/reagents/medicine_reagents.dm

```diff
@@ -706,51 +706,57 @@
 
 /datum/reagent/medicine/strange_reagent/reaction_mob(mob/living/M, method=TOUCH, reac_volume)
-	if(M.stat == DEAD)
-		if(M.suiciding || M.hellbound) //they are never coming back
-			M.visible_message("<span class='warning'>[M]'s body does not react...</span>")
-			return
-		if(M.getBruteLoss() >= 100 || M.getFireLoss() >= 100 || HAS_TRAIT(M, TRAIT_HUSK)) //body is too damaged to be revived
-			M.visible_message("<span class='warning'>[M]'s body convulses a bit, and then falls still once more.</span>")
-			M.do_jitter_animation(10)
-			return
-		else
-			M.visible_message("<span class='warning'>[M]'s body starts convulsing!</span>")
-			M.notify_ghost_cloning(source = M)
-			M.do_jitter_animation(10)
-			addtimer(CALLBACK(M, /mob/living/carbon.proc/do_jitter_animation, 10), 40) //jitter immediately, then again after 4 and 8 seconds
-			addtimer(CALLBACK(M, /mob/living/carbon.proc/do_jitter_animation, 10), 80)
-			sleep(100) //so the ghost has time to re-enter
-
-
-			if(iscarbon(M))
-				var/mob/living/carbon/C = M
-				if(!(C.dna && C.dna.species && (NOBLOOD in C.dna.species.species_traits)))
-					C.blood_volume = max(C.blood_volume, BLOOD_VOLUME_NORMAL) //so you don't instantly re-die from a lack of blood
-				for(var/organ in C.internal_organs)
-					var/obj/item/organ/O = organ
-					O.setOrganDamage(0) //so you don't near-instantly re-die because your heart has decayed to the point of complete failure
-
-			M.adjustOxyLoss(-20, 0)
-			M.adjustToxLoss(-20, 0)
-			M.updatehealth()
-			if(M.revive(full_heal = FALSE, admin_revive = FALSE))
-				M.emote("gasp")
-				log_combat(M, M, "revived", src)
+	if(M.stat != DEAD || method != TOUCH)
+		return ..()
+	if(M.suiciding || M.hellbound) //they are never coming back
+		M.visible_message("<span class='warning'>[M]'s body does not react...</span>")
+		return
+	var/amount_to_revive = round((M.getBruteLoss()+M.getFireLoss())/20)//upto 10 units
+	if(M.getBruteLoss()+M.getFireLoss() >= 200 || HAS_TRAIT(M, TRAIT_HUSK) || reac_volume < amount_to_revive) //body will die from brute+burn on revive or you haven't provided enough to revive.
+		M.visible_message("<span class='warning'>[M]'s body convulses a bit, and then falls still once more.</span>")
+		M.do_jitter_animation(10)
+		return
+	M.visible_message("<span class='warning'>[M]'s body starts convulsing!</span>")
+	M.notify_ghost_cloning(source = M)
+	M.do_jitter_animation(10)
+	addtimer(CALLBACK(M, /mob/living/carbon.proc/do_jitter_animation, 10), 40) //jitter immediately, then again after 4 and 8 seconds
+	addtimer(CALLBACK(M, /mob/living/carbon.proc/do_jitter_animation, 10), 80)
+	sleep(100) //so the ghost has time to re-enter
+
+	var/excess_healing = 5*(reac_volume-amount_to_revive) //excess reagent will heal blood and organs across the board
+	if(iscarbon(M) && excess_healing)
+		var/mob/living/carbon/C = M
+		if(!(C.dna?.species && (NOBLOOD in C.dna.species.species_traits)))
+			C.blood_volume += (excess_healing*2)//1 excess = 10 blood
```

Comment:
Can this cause someone to be gibbed due to having too much blood in them?

Reply:
It would take 795u strange reagent i think. 

The blood healing is weaksauce, not enough to matter in most circumstances.